src/builder/commands/pip.rs

```diff
@@ -326,11 +341,18 @@ fn version_req(hash: &mut Digest, fname: &Path) -> Result<(), VersionError> {
                 if chunk.len() == 0 || chunk.starts_with("#") {
                     continue;
                 }
+                if let Some(req) = parse_req_filename(chunk) {
+                    included_files.push(req);
+                }
                 // Should we also ignore the order?
                 hash.item(chunk);
             }
             Ok(())
-    }).map_err(err)
+    }).map_err(err));
+    for req in &included_files {
```

Comment:
For me this variant is more readable (only minus is dublicating of map_err fn):
```
fn version_req(hash: &mut Digest, fname: &Path) -> Result<(), VersionError> {
    let path = Path::new("/work").join(fname);
    let f = try!(File::open(&path)
                      .map_err(|e| VersionError::Io(e, path.clone())));

    let f = BufReader::new(f);
    for line in f.lines() {
        let line = try!(line.map_err(|e| VersionError::Io(e, path.clone())));
        let chunk = line[..].trim();
        // Ignore empty lines and comments
        if chunk.len() == 0 || chunk.starts_with("#") {
            continue;
        }
        if let Some(req) = parse_req_filename(chunk) {
            try!(version_req(hash, &fname.parent().unwrap().join(req)));
        }
        // Should we also ignore the order?
        hash.item(chunk);
    }
    Ok(())
}
```

than this style
```
fn version_req(hash: &mut Digest, fname: &Path) -> Result<(), VersionError> {
    let path = Path::new("/work").join(fname);
    let err = |e| VersionError::Io(e, path.clone());
    let mut included_files: Vec<String> = Vec::new();
    try!(File::open(&path)
         .and_then(|f| {
            let f = BufReader::new(f);
            for line in f.lines() {
                let line = try!(line);
                let chunk = line[..].trim();
                // Ignore empty lines and comments
                if chunk.len() == 0 || chunk.starts_with("#") {
                    continue;
                }
                if let Some(req) = parse_req_filename(chunk) {
                    included_files.push(req.to_string());
                }
                // Should we also ignore the order?
                hash.item(chunk);
            }
            Ok(())
    }).map_err(err));
    for req in &included_files {
        try!(version_req(hash, &fname.parent().unwrap().join(req)));
    }
    Ok(())
}
```
I can change it if you like.